Feat/email notification

adrianmoloce opened the pull request:
This pull request contains:
- added new dependency httpx
- added send_email as optional paramater to the /invoice endpoint and if set to true a mandatory parameter in the address of the receiver
- created unit tests for those 3 scenarios

amoloce committed:
feat: Add httpx to dependency group.

amoloce committed:
feat: Add optional send_mail parameter.

amoloce committed:
feat: Log a message for requests with send_mail set to true

amoloce committed:
feat: Handle exception address not provided with send_mail true

- Created handler for exception where address is not provided but
  send_mail is set to True
- Updated and created new unit tests to test functionality

adrianmoloce commented on src/invoice_utils/web.py:
```diff
@@ -36,14 +42,33 @@ class InvoiceEntity(BaseModel):
+    items: list[InvoicedItem]
+
+
+class InvoicingRequestInputError(Exception):
```
Change to InvoiceRequestInputError for consistency.

koosie0507 reviewed (approved):
Pretty good for a start

amoloce committed:
Send email notification with smtplib. (#5)

* feat: Add pytest-mock and python-dotenv as dependencies

* feat: Created config file with environment variables

* feat: Send email notification with smtplib if send_mail

- Send email with smtplib if send_mail flag is true
- Created test for sending the mail and smtplib exceptions
- Created custom InvoiceRequestEmailError

* feat: Add note for setting up env variables in README

* feat: Add namespace and default values to environment variables.

* Feat: add basic configurations for sending emails. (#7)

* feat: Config mail subject

* feat: SMTP login

- log into SMTP

* feat: Make SMTP Login Optional

* feat: Introduce SMTP start_tls optional env flag

* fix: Increase resilience of smtp tls flag

- method to convert string to bool for any env flag
- added tests cases for the tls flag

amoloce committed:
Merge branch 'main' into feat/email-notification
